plugins/autocast/src/autocast.plugin.ts

```diff
@@ -14,50 +17,52 @@ export function autocast(
   }
 ) {
   return (listener: FlatfileListener) => {
-    listener.on(
-      'commit:created',
-      { sheetSlug },
-      async (event: FlatfileEvent) => {
-        await BulkRecordHook(
-          event,
-          async (records: FlatfileRecord[]) => {
-            const sheetId = event.context.sheetId
-            const sheet = await api.sheets.get(sheetId)
-            if (!sheet) {
-              logInfo('@flatfile/plugin-autocast', 'Failed to fetch sheet')
-              return
-            }
+    listener.use(
+      bulkRecordHookPlugin(
+        sheetSlug,
+        async (records: FlatfileRecord[], event: FlatfileEvent) => {
+          const sheetId = event.context.sheetId
+          const sheet = await api.sheets.get(sheetId)
+          if (!sheet) {
+            logInfo('@flatfile/plugin-autocast', 'Failed to fetch sheet')
+            return
+          }
 
-            const castableFields = sheet.data.config.fields.filter((field) =>
-              fieldFilters
-                ? fieldFilters.includes(field.key)
-                : field.type !== 'string'
-            )
-            records.forEach((record) => {
-              castableFields.forEach((field) => {
-                const originalValue = record.get(field.key)
-                const caster = CASTING_FUNCTIONS[field.type]
+          const castableFields = sheet.data.config.fields.filter((field) =>
+            fieldFilters
+              ? fieldFilters.includes(field.key)
+              : field.type !== 'string'
+          )
+          records.forEach((record) => {
+            castableFields.forEach((field) => {
+              const originalValue = record.get(field.key)
+              const caster = CASTING_FUNCTIONS[field.type]
 
-                if (
-                  originalValue &&
-                  caster &&
-                  typeof originalValue !== field.type
-                ) {
-                  try {
-                    record.computeIfPresent(field.key, caster)
-                  } catch (e) {
-                    record.addError(
-                      field.key,
-                      e.message || 'Failed to cast value'
-                    )
-                  }
+              if (
+                originalValue &&
+                caster &&
+                typeof originalValue !== field.type
+              ) {
+                try {
+                  record.computeIfPresent(
+                    field.key,
+                    caster,
+                    `Cast '${originalValue}' from '${typeof originalValue}' to '${
+                      field.type
+                    }'`
```

Comment:
Note: This seems to be overridden somewhere

Reply:
Nvm...it is additive:
![image](https://github.com/FlatFilers/flatfile-plugins/assets/4754531/918eeb28-7ec7-4848-8669-58962572b77c)